tensorflow/core/kernels/data/optimize_dataset_op_test.cc

```diff
@@ -45,66 +45,221 @@ class OptimizeDatasetOpTest : public DatasetOpsTestBase {
+    Tensor stop = CreateTensor<int64>(TensorShape({}), {3});
+    Tensor step = CreateTensor<int64>(TensorShape({}), {1});
+    TF_RETURN_IF_ERROR(
+        RunFunction(test::function::MakeRangeDataset(),
+                    /*attrs*/
+                    {{"output_types", DataTypeVector({DT_INT64})},
+                     {"output_shapes",
+                      std::vector<PartialTensorShape>{PartialTensorShape({})}}},
+                    /*inputs*/ {start, stop, step}, graph_opts,
+                    /*rets*/ {&range_dataset_tensor}));
+
+    if (dataset_type == "TakeDataset") {
+      Tensor count_tensor = CreateTensor<int64>(TensorShape({}), {count});
+      TF_RETURN_IF_ERROR(RunFunction(
+          test::function::MakeTakeDataset(),
+          /*attrs*/
+          {{"output_types", output_types}, {"output_shapes", output_shapes}},
+          /*inputs*/ {range_dataset_tensor, count_tensor}, graph_opts,
+          /*rets*/ {input_dataset}));
+    } else if (dataset_type == "SkipDataset") {
+      Tensor count_tensor = CreateTensor<int64>(TensorShape({}), {count});
+      TF_RETURN_IF_ERROR(RunFunction(
+          test::function::MakeSkipDataset(),
+          /*attrs*/
+          {{"output_types", output_types}, {"output_shapes", output_shapes}},
+          /*inputs*/ {range_dataset_tensor, count_tensor}, graph_opts,
+          /*rets*/ {input_dataset}));
+    } else if (dataset_type == "PrefetchDataset") {
+      Tensor buffer_size = CreateTensor<int64>(TensorShape({}), {count});
+      TF_RETURN_IF_ERROR(RunFunction(
+          test::function::MakePrefetchDataset(),
+          /*attrs*/
+          {{"output_types", output_types},
+           {"output_shapes", output_shapes},
+           {"slack_period", 0}},
+          /*inputs*/ {range_dataset_tensor, buffer_size}, graph_opts,
+          /*rets*/ {input_dataset}));
+    } else if (dataset_type == "RepeatDataset") {
+      Tensor count_tensor = CreateTensor<int64>(TensorShape({}), {count});
+      TF_RETURN_IF_ERROR(RunFunction(
+          test::function::MakeRepeatDataset(),
+          /*attrs*/
+          {{"output_types", output_types}, {"output_shapes", output_shapes}},
+          /*inputs*/ {range_dataset_tensor, count_tensor}, graph_opts,
+          /*rets*/ {input_dataset}));
+    }
+
+    return Status::OK();
+  }
+};
+
+struct NoopEliminationTestCase {
+  string dataset_type;
+  int64 count;
+  DataTypeVector output_types;
+  std::vector<PartialTensorShape> output_shapes;
+  std::vector<Tensor> expected_outputs;
 };
 
-TEST_F(OptimizeDatasetOpTest, FilterFusion) {
+template <typename T>
+std::vector<Tensor> ConvertToTensorVec(std::vector<T> values) {
+  std::vector<Tensor> tensors;
+  tensors.reserve(values.size());
+  for (auto& value : values) {
+    tensors.emplace_back(
+        DatasetOpsTestBase::CreateTensor<T>(TensorShape({}), {value}));
+  }
+  return tensors;
+}
+
+NoopEliminationTestCase NoopEliminationTestCase1() {
```

Comment:
you do not need to test all the different types of noop eliminitations ... the purpose of this unit test is to validate that the dataset op kernel that performs optimization works and using it with a single noop elimination would be sufficient

Reply:
Gotcha. Just remove the other test cases and rewrite the noop_elimination test via this commit (https://github.com/tensorflow/tensorflow/pull/28874/commits/38d62f834f4f4806d5c86174852b7a2eb6a84d63).